File: e2e/playwright-typescript/interactions/steps/AccessibilityStep.ts
import { Page } from '@playwright/test';
import AxeBuilder from '@axe-core/playwright';
import type { AxeResults } from 'axe-core';
import { step } from '@/decorators/step-decorator';

export class AccessibilityStep {
  constructor(private readonly page: Page) {}

  @step('Analyze page accessibility with axe')
  async analyze(): Promise<AxeResults> {
    return new AxeBuilder({ page: this.page }).analyze();
  }
}

File: e2e/playwright-python/interactions/steps/accessibility_step.py
from playwright.sync_api import Page
from axe_playwright_python.sync_playwright import Axe
from decorators.step_decorator import step


class AccessibilityStep:
    def __init__(self, page: Page):
        self._page = page
        self._axe = Axe()

    @step("Analyze page accessibility with axe")
    def analyze(self, selector: str | None = None):
        context = {"include": [[selector]]} if selector else None
        return self._axe.run(self._page, context=context)
